### PR TITLE
Add test to cover nil gin.Context case

### PR DESCRIPTION
**Pull request by tomaisthorpe**

### CODE REVIEW

File: instrumentation/sources/gin-gonic/gin/middleware.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 func GetMiddleware() gin.HandlerFunc {
 	return func(c *gin.Context) {
 		if c == nil {
-			return // Don't investigate empty requests.
+			return // Gin should never pass us a nil context
```
[the remark this answers is not in the document]
Comment (tomaisthorpe) [reply instrumentation/sources/gin-gonic/gin/middleware.go:14]: It's a fair point, but would rather just handle the potential panic.
